packages/core/src/internal/sagaHelpers/retry.js

```diff
@@ -0,0 +1,29 @@
+import fsmIterator, { qEnd } from './fsmIterator'
+import { call, delay } from '../io'
+
+export default function retry(maxTries, delayLength = 0, fn, ...args) {
+  let counter = maxTries
+
+  const yCall = ({ done: false, value: call(fn, ...args) })
+  const yDelay = { done: false, value: delay(delayLength) }
+
+  return fsmIterator(
+    {
+      q1() {
+        return ['q2', yCall, null, 'q10']
```

Comment:
we could refactor it to object to simplify signature

Reply:
personally i kinda like tuple-approach, it's really concise - however this is not using it "correctly", tuples should be of fixed size and we return different lengths here - also [accessing out of bounds indexes](https://github.com/redux-saga/redux-saga/pull/1430/files#diff-bcf10a3064550fa33b1534dcf38328f0R37) is a v8 deopt and should be avoided

to sum it up - i agree that we should refactor it to object, i propose `{ nextQ, errQ, eff, updater }`, although we could also go with `{ nextState, errorState, effect, stateUpdater }`, wdyt?

Reply:
👍 `{ nextState, errorState, effect, stateUpdater }`

Reply:
Sure thing!